### ISSUE TITLE
MongoDB: Cannot create m-n relations

### ISSUE THREAD

**Issue by killiandvcz** [prisma/prisma#10071]
### Bug description

I cannot create Many to Many relations (MongoDB). I followed the documentation, but this leads to the following error:
```
Prisma schema loaded from prisma/schema.prisma
Error: Schema parsing
error: Error parsing attribute "@relation": The type of the field `categoryIDs` in the model `Post` is not matching the type of the referenced field `id` in model `Category`.
  -->  schema.prisma:14
   | 
13 |   categoryIDs String[]   @db.Array(ObjectId)
14 |   categories  Category[] @relation(fields: [categoryIDs])
15 | }
   | 
error: Error parsing attribute "@relation": The type of the field `postIDs` in the model `Category` is not matching the type of the referenced field `id` in model `Post`.
  -->  schema.prisma:20
   | 
19 |   postIDs String[] @db.Array(ObjectId)
20 |   posts   Post[]   @relation(fields: [postIDs])
21 | }
   | 

Validation Error Count: 2
```

### How to reproduce

1. Create a prisma schema with the example of the [documentation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#mongodb)
2. npx prisma generate

### Prisma information

```
datasource db {
  provider = "mongodb"
  url      = "********"
}

generator client {
  provider = "prisma-client-js"
  previewFeatures = ["mongoDb"]
}

model Post {
  id          String     @id @default(dbgenerated()) @map("_id") @db.ObjectId
  categoryIDs String[]   @db.Array(ObjectId)
  categories  Category[] @relation(fields: [categoryIDs])
}

model Category {
  id      String   @id @default(dbgenerated()) @map("_id") @db.ObjectId
  postIDs String[] @db.Array(ObjectId)
  posts   Post[]   @relation(fields: [postIDs])
}

```

### Environment & setup

- OS: Windows with WSL2 (Ubuntu 20.04)
- Database: MongoDB
- Node.js version: 14.18.1

### Prisma Version

```
prisma                  : 3.4.0
@prisma/client          : 3.4.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 1c9fdaa9e2319b814822d6dbfd0a69e1fcc13a85 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 1c9fdaa9e2319b814822d6dbfd0a69e1fcc13a85 (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core 1c9fdaa9e2319b814822d6dbfd0a69e1fcc13a85 (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt 1c9fdaa9e2319b814822d6dbfd0a69e1fcc13a85 (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : 1c9fdaa9e2319b814822d6dbfd0a69e1fcc13a85
Studio                  : 0.438.0
Preview Features        : mongoDb
```

**Comment by dannysofftie** [comment]
Bumped into this as well.

To give more clarification on how this happened, the errors only show after upgrading to Prisma v3.4.0, Prisma v3.3.0 runs perfectly fine.

The option, for now, is to downgrade to v3.3.0
1. Remove `prisma` & `@prisma/client` - `yarn remove prisma @prisma/client`
2. Install `v3.3.0` - `yarn add prisma@3.3.0 -D && yarn add @prisma/client@3.3.0`

**Comment by killiandvcz** [comment]
Yes, indeed there are no more errors in the generation. However, I had to create [another issue](https://github.com/prisma/studio/issues/797) for Prisma Studio. Even in v3.3.0, cannot correctly use Many to many relationships.

**Comment by reslear** [comment]
same problem @dannysofftie and downgrade ext to 3.3.0
<img width="745" alt="Screenshot 2021-11-03 at 19 50 56" src="https://user-images.githubusercontent.com/12596485/140106968-2097545a-bc88-46c8-9fc2-401e1992a855.png">

**Comment by dannysofftie** [comment]
> same problem @dannysofftie and downgrade ext to 3.3.0

Agreeable

**Comment by reslear** [comment]
hi @matthewmueller, see please this problem

**Comment by dimaip** [comment]
We are hit by this too!
It's a critical blocker, and there are other important bugfixes in 3.4 that we really need.

**Comment by tomhoule** [comment]
We identified the issue and are working on a fix. That work will be tracked there: https://github.com/prisma/prisma/issues/10071

**Comment by Jolg42** [comment]
We just released a patch, you can now use the [`3.4.1`](https://github.com/prisma/prisma/releases/tag/3.4.1) version that includes the fix.
